src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -79,44 +86,41 @@
+      <Output TaskParameter="UniqueItems" ItemName="_TestDependencies" />
+    </RemoveCommonItems>
 
     <!-- Remove duplicates. Note that we mustn't just copy in sequence and let
          the last one win that way because it will cause copies to occur on
          every incremental build. -->
     <ItemGroup>
-      <_TestCopyLocalByFileName Include="@(TestCopyLocal->'%(RecursiveDir)%(FileName)%(Extension)')">
-        <NugetPackageId>%(TestCopyLocal.NugetPackageId)</NugetPackageId>
+      <_TestDependenciesByFileName Include="@(_TestDependencies->'%(RecursiveDir)%(FileName)%(Extension)')">
+        <NugetPackageId>%(_TestDependencies.NugetPackageId)</NugetPackageId>
         <SourcePath>%(Identity)</SourcePath>
-      </_TestCopyLocalByFileName>
+      </_TestDependenciesByFileName>
     </ItemGroup>
 
-    <RemoveDuplicatesWithLastOneWinsPolicy Inputs="@(_TestCopyLocalByFileName)">
-      <Output TaskParameter="Filtered" ItemName="_TestCopyLocalByFileNameWithoutDuplicates" />
+    <RemoveDuplicatesWithLastOneWinsPolicy Inputs="@(_TestDependenciesByFileName)">
+      <Output TaskParameter="Filtered" ItemName="_TestDependenciesByFileNameWithoutDuplicates" />
     </RemoveDuplicatesWithLastOneWinsPolicy>
 
-    <!-- Ideally, we'd have UseHardLinksIfPossible on by default because we copy tons of the
-         same files for every single test project here, but it doesn't currently work x-plat.
-         so we only turn it on by default for Windows builds for now. -->
-    <PropertyGroup>
-      <CreateHardLinksForCopyTestToTestDirectoryIfPossible Condition="'$(CreateHardLinksForCopyTestToTestDirectoryIfPossible)'=='' and '$(OS)' == 'Windows_NT'">true</CreateHardLinksForCopyTestToTestDirectoryIfPossible>
-      <CreateHardLinksForCopyTestToTestDirectoryIfPossible Condition="'$(CreateHardLinksForCopyTestToTestDirectoryIfPossible)'==''">$(CreateHardLinksForCopyFilesToOutputDirectoryIfPossible)</CreateHardLinksForCopyTestToTestDirectoryIfPossible>
-    </PropertyGroup>
-
     <ItemGroup>
-      <SourcesToCopyToTestDir Include="@(RunTestsForProjectInputs)" />
-      <DestinationsForTestDir Include="@(RunTestsForProjectInputs->'$(TestPath)\%(Filename)%(Extension)')" />
-
-      <SourcesToCopyToTestDir Include="@(ContentWithTargetPath)" />
-      <DestinationsForTestDir Include="@(ContentWithTargetPath->'$(TestPath)\%(TargetPath)')" />
-
-      <SourcesToCopyToTestDir Include="@(OverridenRuntimeContent)" />
-      <DestinationsForTestDir Include="@(OverridenRuntimeContent->'$(TestPath)\%(Filename)%(Extension)')" />
+      <!-- Not all platforms can use the .ni.dlls that come from packages.  If TestWithoutNativeImages is specified, we'll exclude them from copy generation.
+           If we end up needing this for any other sorts of filtering, we'll want to add a list of filtered extensions to be matched on EndsWith.  -->
+      <IncludedFileForRunnerScript Include="@(_TestDependenciesByFileNameWithoutDuplicates)"
```

Comment:
Without digging deep I cannot tell if _TestDependenciesByFileNameWithoutDuplicates is the same set of items as we are copying below via SourcesToCopyToTestDir. I would really like the msbuild copy and the script copy item groups to be exactly the same so that is clear.